label_studio/tests/data_manager/test_columns_api.py

```diff
@@ -0,0 +1,84 @@
+import json
+import pytest
+
+from rest_framework import status
+from pprint import pprint
+from ..utils import project_id
+
+pytestmark = pytest.mark.django_db
+
+
+def _get_columns(business_client, label_config=None):
+    r = business_client.post(
+        "/api/projects/",
+        data=json.dumps(
+            dict(
+                title="test_project1",
+                **({'label_config': label_config} if label_config else {})
+            )
+        ),
+        content_type="application/json",
+    )
+
+    project1_id = r.json()["id"]
+    r = business_client.get(f"/api/dm/columns/?project={project1_id}")
+
+    assert r.status_code == 200, r.content
+    r_json = r.json()
+    assert 'columns' in r_json
+    return r_json['columns']
+
+
+def test_columns_api_returns_expected_ids(business_client):
+    columns = _get_columns(business_client)
+
+    assert [c['id'] for c in columns] == [
+        'id',
+        'inner_id',
+        'completed_at',
+        'total_annotations',
+        'cancelled_annotations',
+        'total_predictions',
+        'annotators',
+        'annotations_results',
+        'annotations_ids',
+        'predictions_score',
+        'predictions_model_versions',
+        'predictions_results',
+        'file_upload',
+        'storage_filename',
+        'created_at',
+        'updated_at',
+        'updated_by',
+        'avg_lead_time',
+        'data',
+    ]
+
+
+def test_columns_api_annotates_default_columns_with_project_defined_false(business_client):
+    columns = _get_columns(business_client)
+
+    for c in columns:
+        assert 'project_defined' in c
+        assert c['project_defined'] is False
+
+
+def test_columns_api_annotates_config_defined_columns_with_project_defined_true(business_client):
+    config_with_text_column = (
+        """
+        <View>
+            <Text value="$text" name="artist" />
```

Comment:
Could we add a specific case for the original prevalent issue created with the `$data` usage as well.

Reply:
The original issue manifests only in LSE, it'll be specifically tested there (see also, how this change is used to fix the issue: https://github.com/HumanSignal/label-studio-enterprise/pull/4411/files#diff-a3d9c0b437f32cd5dca176d0a4268547afa904f30ec22fd7cee07ec09cf86130R18-R19)